hub/core/io.py

```diff
@@ -405,16 +411,18 @@ def _use_cache(self, storage: Union[StorageProvider, LRUCache]) -> LRUCache:
 
     def _map_chunk_engines(self, tensors: Sequence[str]) -> Dict[str, ChunkEngine]:
         return {
-            key: self._create_chunk_engine(key, self.dataset.version_state)
-            for key in tensors
+            name: self._create_chunk_engine(name, self.dataset.version_state)
+            for name in tensors
         }
 
-    def _create_chunk_engine(self, tensor_key, version_state):
+    def _create_chunk_engine(self, tensor_name, version_state):
+        tensor_key = version_state["tensor_names"][tensor_name]
         return ChunkEngine(tensor_key, self._use_cache(self.storage), version_state)
 
     def _get_dataset_indicies(self):
+        version_state = self.dataset.version_state
         tensor_lengths = [
-            len(self.dataset.version_state["full_tensors"][tensor])
+            len(version_state["full_tensors"][version_state["tensor_names"][tensor]])
```

Comment:
there should be a better way of doing this, seems very unnecessary.

Reply:
I think it can just be `self.dataset[tensor]` but then is there any reason why version_state was used in the first place?